src/core/gui/MainWindow.cpp

```diff
@@ -82,9 +82,9 @@ MainWindow::MainWindow(GladeSearchpath* gladeSearchPath, Control* control, GtkAp
 
     g_signal_connect(get("buttonCloseSidebar"), "clicked", G_CALLBACK(buttonCloseSidebarClicked), this);
 
-
-    // "watch over" all events
-    g_signal_connect(this->window, "key-press-event", G_CALLBACK(onKeyPressCallback), this);
+    // "watch over" all key events
+    g_signal_connect(this->window, "key-press-event", G_CALLBACK(gtk_window_propagate_key_event), nullptr);
+    g_signal_connect(this->window, "key-release-event", G_CALLBACK(gtk_window_propagate_key_event), nullptr);
```

Comment:
Are those not already the default callbacks? The `gtk_window_propagate_key_event` [doc](https://docs.gtk.org/gtk3/method.Window.propagate_key_event.html) says
> This is normally called by the default ::key_press_event and ::key_release_event handlers for toplevel windows

Could we maybe remove those two lines then?

Reply:
I also thought that would mean we can remove these two lines. It's not true though, they make a difference. Plugin accelerators are called before the text editor can handle the input without those lines. So we must leave them.

Reply:
Here's the explanation (see https://docs.gtk.org/gtk3/input-handling.html):

> For key events, the top-level window’s default [GtkWidget::key-press-event](https://docs.gtk.org/gtk3/signal.Widget.key-press-event.html) and [GtkWidget::key-release-event](https://docs.gtk.org/gtk3/signal.Widget.key-release-event.html) signal handlers handle mnemonics and accelerators first. Other key presses are then passed to [gtk_window_propagate_key_event()](https://docs.gtk.org/gtk3/method.Window.propagate_key_event.html) which propagates the event upwards from the window’s current focus widget ([gtk_window_get_focus()](https://docs.gtk.org/gtk3/method.Window.get_focus.html)) to the top-level.